src/hooks/useUniversalRouter.ts

```diff
@@ -58,6 +58,8 @@ export function useUniversalRouterSwapCallback(
         if (!chainId) throw new Error('missing chainId')
         if (!provider) throw new Error('missing provider')
         if (!trade) throw new Error('missing trade')
+        const connectedChainId = await provider.getSigner().getChainId()
+        if (chainId !== connectedChainId) throw new Error('signer chainId does not match')
```

Comment:
Will this bubble up to Sentry?

Reply:
no currently we aren't logging general swap failures to sentry